src/Implement/UnitTestDownloadStream.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "DownloadStream.h"
+
+// test basic usage
+TEST(DOWNLOADSTREAM_TEST, Basic) {
+  // create temporary test file
+  std::fstream newfile("test", std::fstream::out);
+  newfile << "123456";
```

Comment:
Please make the test string a constant. i.e. `constexpr char kTestFileContent[]`, this is so that we've the flexibility of changing the test string.
Also, consider adding characters such as '\0' and '\n' into the test string, this is a common corner case for many string related code.

Reply:
done. Assuming the file is plaintext, there won't be '\0' in the middle of file content.